### PR TITLE
(DOCSP-18612): Add sync production load testing doc

### PR DESCRIPTION
**Pull request by cbush**
### Jira

- https://jira.mongodb.org/browse/DOCSP-18612

### Staged Changes (Requires MongoDB Corp SSO)

- https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/sync-load-testing/sync/production-load-testing/

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sync/production-load-testing.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+The exact details of the simulation depend on your app's actual
+workload.
+
+.. note::
+
+   The Node.js SDK uses the same underlying C++ database and sync client
+   as the mobile (iOS, Android, .NET, etc.) SDKs. Therefore, you can
+   exercise the same code paths as in your production mobile app while
+   benefiting from the headless/server environment of Node.js.
```
Comment (MongoCaleb) [inline source/sync/production-load-testing.txt:40]: Is there supposed to be a slash in headless/server??

File: source/sync/production-load-testing.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+To **load test** is to simulate the requests that users will put on your
+app once your app is in production. The goal is to measure performance
+and to identify and resolve issues before they happen to a real user.
+
+Recommended Strategy
+--------------------
+
+Use the Node.js SDK to simulate a common single user's workload for your
+mobile app. For example:
```
Comment (MongoCaleb) [inline source/sync/production-load-testing.txt:27]: For example, you can use the SDK to:

File: source/sync/production-load-testing.txt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   exercise the same code paths as in your production mobile app while
+   benefiting from the headless/server environment of Node.js.
+
+You can then deploy the single-user simulation to hundreds or thousands
+of concurrent nodes in order to test a real multi-user workload. For example:
+
+- Package the app in a Docker container.
+- Deploy the container as part of a Kubernetes job.
+- Adjust the parameters of the job to match it to a real production workload.
+- Measure the time it takes to complete the Kubernetes job to calculate overall performance.
```
Comment (MongoCaleb) [inline source/sync/production-load-testing.txt:48]: Do we want a link to a place to get more info on Kubernetes? Or a link in the final sentence about any recommendations? Too risky, I suppose.

**Review by MongoCaleb — APPROVED** [review]
LGTM % a few rewording suggestions.

File: source/sync/production-load-testing.txt (hunk 3 above)
Comment (cbush) [reply source/sync/production-load-testing.txt:48]: Yeah, I decided against it. I think this guide is for an advanced-enough user who can figure that part out for themselves. This doc is just trying to give guidance.

File: source/sync/production-load-testing.txt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+app once your app is in production. The goal is to measure performance
+and to identify and resolve issues before they happen to a real user.
+
+Recommended Strategy
+--------------------
+
+Use the Node.js SDK to simulate a common single user's workload for your
+mobile app. For example, you can use the SDK to:
+
+- Open a set number of {+realms+}.
```
Comment (ianpward) [inline source/sync/production-load-testing.txt:29]: might want to say you are opening realms that already have initialized sync data on the server-side to simulate read use cases. Then maybe open different realms to simulate write workloads.

File: source/sync/production-load-testing.txt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- Make a certain number of writes to each partition.
+
```
Comment (ianpward) [inline source/sync/production-load-testing.txt:31]: May also want to add that you should simulate any backend server API calls on the Atlas cluster at the same ie. from a driver.

File: source/sync/production-load-testing.txt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+- Open a set number of {+realms+}.
+- Make a certain number of writes to each partition.
+- Simulate backend server calls (for example, by calling Realm functions).
+
+The exact details of the simulation depend on your app's actual
+workload. For instance, you might open {+realms+} that already have
+initialized sync data on the server side to simulate read use cases. You
+might open different {+realms+} to simulate write workloads. Writer
+realms might be a mix of shared or private per-user realms depending on
```
Comment (MongoCaleb) [inline source/sync/production-load-testing.txt:37]: nit: I know that "writer realms" is a term Ian used in his comments, but I've not seen it used anywhere else. Not sure if that's an official term we want to use.

**Review by MongoCaleb — APPROVED** [review]
1 wee nit, laddie.

File: source/sync/production-load-testing.txt (hunk 6 above)
Comment (cbush) [reply source/sync/production-load-testing.txt:37]: No, it's meant to be "These realms" (referring to what was set up in last sentence). Updated.

**Pull request merged**